Add config option to not teleport on free

Lazersmoke opened the pull request:
This was in prison pearl, but was lost in the switch to exile pearl. Related: CivClassic/Configs#22

Lazersmoke committed:
Add config option to not teleport on free

TealNerd commented:
this will keep the players in the end, move the check to where it actually teleports players

Lazersmoke commented:
Yeah, that's the intent, and how it worked with prison pearl. You take the end portal out or die and it respawns you as normal. It also prevents your inventory from getting dropped.

Do you want it like this:
```
if(pearl.getPearlType() == PearlType.PRISON) {
  dropInventory(player);
  if(pearlApi.getPearlConfig().getShouldFreeTeleport() && (reason == PearlFreeReason.FREED_BY_PLAYER || reason == PearlFreeReason.PEARL_THROWN)) {
    player.teleport(pearl.getLocation().add(0, 0.5, 0));
  } else {
    SpawnUtil.spawnPlayer(player);
  }
}
```

So you get random spawned instead of teleported to your imprisoner?

Lazersmoke commented:
@TealNerd

Lazersmoke committed:
Randomspawn on /ep free

Previously, players would stay in the end if `/ep free`d with free teleport disabled.
